File: app/src/main/java/com/esorokin/lantector/model/network/data/ApiStatusResult.java
package com.esorokin.lantector.model.network.data;

import android.text.TextUtils;

public enum ApiStatusResult {
	SUCCESS("OK"),
	ERROR("ERROR");

	private String code;

	ApiStatusResult(String code) {
		this.code = code;
	}

	public static ApiStatusResult from(String code) {
		for (ApiStatusResult apiStatusResult : ApiStatusResult.values()) {
			if (TextUtils.equals(apiStatusResult.code, code)) {
				return apiStatusResult;
			}
		}

		return ERROR;
	}
}

File: app/src/main/java/com/esorokin/lantector/model/network/data/ApiStatusResult.kt
package com.esorokin.lantector.model.network.data

enum class ApiStatusResult(val code: String = "") {
    SUCCESS("OK"),
    ERROR("ERROR");

    companion object {
        fun from(code: String): ApiStatusResult {
            ApiStatusResult.values().forEach { if (it.code == code) return it }
            return ERROR
        }
    }
}
